### ISSUE TITLE
Fully convolutional network variable input size

### ISSUE THREAD

**Issue by adaber** [Microsoft/CNTK#2321]
Hi,

As far as I know, CNTK convolutional networks have fixed size inputs. I was wondering how feasible/possible it would be to implement a logic in CNTK that would allow us to create fully convolutional networks with variable size inputs (variable input image size) ?

I am not suggesting to be able to use different size images within the same training session(minibatch). I know that it is inefficient (if not impossible). I am talking about using the same fully convolutional neural net model on variable size images where a training set consists of images of the same size.

Thanks!

**Comment by n17s** [comment]
This is being fixed as we speak. Convolutional layers now support variable size inputs and pooling layers will support it soon. You need to define your input variables as having free dimensions e.g. in Python 
```python
img = C.input_variable(shape=(3, C.FreeDimension, C.FreeDimension))
```
We are aiming to get this out for the 2.2 release next week.

**Comment by adaber** [comment]
That's great news :) How would you define a "free dimension" input variable in C++ ?

Thanks!

**Comment by adaber** [comment]
Sorry for the double post. I can't seem to be able to edit my previous post.

I've coded all of my "trainers" in C++ and so it'd be great if you could let me know some of the differences in creating a minibatch and other related things now that CNTK will support a variable size convolutional layers ? Thanks!

**Comment by cha-zhang** [comment]
As of v2.2 free static axis for arbitrary input image size should have been fully supported. Close this issue now.
